assets/js/common/session.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+import $ from 'jquery';
+import debounce from 'lodash/debounce';
+import { trackEvent } from '../helpers/metrics';
+
+const expiryCheckIntervalSeconds = 30; // The interval we'll check timeouts in
+const warningShownSecondsRemaining = 5 * 60; // The time before logout we'll show a warning at
+
+function handleSessionExpiration() {
+    let sessionInterval;
+    let isAuthenticated = true;
+    let expiryTimeRemaining = window.AppConfig.sessionExpirySeconds;
+
+    function startSessionExpiryWarningTimer() {
+        sessionInterval = window.setInterval(
+            sessionTimeoutCheck,
+            expiryCheckIntervalSeconds * 1000
+        );
+    }
+
+    function clearSessionExpiryWarningTimer() {
+        window.clearInterval(sessionInterval);
+    }
+
+    function sessionTimeoutCheck() {
+        expiryTimeRemaining = expiryTimeRemaining - expiryCheckIntervalSeconds;
+        if (expiryTimeRemaining <= 0) {
+            // The user's cookie has expired
+            isAuthenticated = false;
+            clearSessionExpiryWarningTimer();
+            trackEvent('Session', 'Warning', 'Timeout reached');
+        } else if (expiryTimeRemaining <= warningShownSecondsRemaining) {
+            // The user has a few minutes remaining before logout
+            trackEvent('Session', 'Warning', 'Timeout almost reached');
+        }
+    }
+
+    const handleActivity = () => {
+        if (isAuthenticated) {
+            // Extend their session
+            $.ajax({
```

Comment:
Does this need `withCredentials` or similar, or does `$.ajax` send through the session cookie with the request by default?

Reply:
When it's using the same domain as the current page, cookies are included by default so we don't need to change anything.